pkg/libvirt/libvirt.go

```diff
@@ -162,6 +162,26 @@ func (d *Driver) setVcpus(cpus uint) error {
 	return nil
 }
 
+func (d *Driver) maybeResizeDiskImage(newCapacity uint64) error {
+	diskCapacity, err := d.getVolCapacity()
+	if err != nil {
+		log.Debugf("failed to get volume capacity")
+		return err
+	}
+
+	if newCapacity == 0 || newCapacity == diskCapacity {
```

Comment:
The second check (`newCapacity == diskCapacity`) is the same as L155 in pkg/libvirt/storage.go given that `capacity` and `diskCapacity` both denote the return value of `d.getVolCapacity()`.

Reply:
This can be a check by itself in a separate function, preventing the name maybe here;

checkIfResizeNeeded(capacity) -> bool, err (needed, failureError)
resizeDiskImage(capacity) -> bool, err  (success, failureError)

Reply:
Fixed.